sympy/solvers/solvers.py

```diff
@@ -948,52 +1016,42 @@ def _sympified_list(w):
                 else Abs(arg))
             if was == fi:
                 break
-
         for e in fi.find(Abs):
             if e.has(*symbols):
                 raise NotImplementedError('solving %s when the argument '
                     'is not real or imaginary.' % e)
-
         # arg
         fi = fi.replace(arg, lambda a: arg(a).rewrite(atan2).rewrite(atan))
-
         # save changes
         f[i] = fi
-
     # see if re(s) or im(s) appear
-    freim = [fi for fi in f if fi.has(re, im)]
-    if freim:
```

Comment:
I have since fixed some performance problems with this section. If it it being changed then the benchmarks should be checked to see if the changes here give any slowdown.

Reply:
@oscarbenjamin I ran the benchmarks. There isn't any slowdown. In fact, certain benchmarks ran faster by 1 or 2 seconds.

Sorry for the late replies. I was busy with my exams.

Reply:
No need to be sorry.

Can you show the benchmark comparison output?

Reply:
Ok, I was a little wrong. I couldn't run the benchmark correctly. These changes are actually resulting in slowdown. The "before" is this commit. The "after" is the master.

```
       before           after         ratio
     [9b6b79a8]       [bf2e1a4e]
     <solve_organise>       <master>  
          423±3ns          418±2ns     0.99  baseline.TimeBaseline.time_loop10
        231±0.1ms        229±0.5ms     0.99  baseline.TimeBaseline.time_loop10000000
         81.1±1ns       80.0±0.5ns     0.99  baseline.TimeBaseline.time_noop
      1.35±0.01μs         1.35±0μs     1.00  baseline.TimeBasicGCD.time_gcd
      4.75±0.01μs      4.74±0.03μs     1.00  core.arit.Time_Arit.time_Add_1x
      33.2±0.03μs       33.0±0.2μs     0.99  core.arit.Time_Arit.time_Add_x05
      4.60±0.03μs      4.69±0.01μs     1.02  core.arit.Time_Arit.time_Add_x1
      2.53±0.04μs      2.62±0.02μs     1.04  core.arit.Time_Arit.time_Add_xy
      1.58±0.01μs      1.59±0.03μs     1.01  core.arit.Time_Arit.time_Add_xyz
      6.06±0.03μs      5.96±0.08μs     0.98  core.arit.Time_Arit.time_Div_2y
      4.02±0.05μs      4.08±0.05μs     1.01  core.arit.Time_Arit.time_Div_xy
      2.59±0.01μs      2.66±0.01μs     1.03  core.arit.Time_Arit.time_Mul_xy
      1.57±0.01μs         1.61±0μs     1.03  core.arit.Time_Arit.time_Mul_xyz
          760±9μs          762±3μs     1.00  core.arit.Time_Arit.time_mod_pow
      2.86±0.02μs      2.80±0.02μs     0.98  core.arit.Time_Arit.time_neg
         845±10ns         833±20ns     0.99  core.arit.Time_Arit.time_pow100
         851±20ns          849±2ns     1.00  core.arit.Time_Arit.time_pow2
       40.0±0.4μs      39.1±0.05μs     0.98  core.arit.Time_Arit.time_pow_im
      13.4±0.04μs      13.7±0.03μs     1.02  core.arit.Time_assumption.time_Add_Mul_is_finite
      19.0±0.05μs      18.9±0.07μs     1.00  core.arit.Time_assumption.time_Add_Mul_is_integer
       51.2±0.3μs       51.1±0.1μs     1.00  core.arit.Time_assumption.time_ncmul
       58.3±0.1μs       57.4±0.3μs     0.98  core.arit.Time_assumption.time_ncpow
       17.7±0.2μs      17.9±0.08μs     1.01  core.expand.TimeExpand.time_expand_arit
      14.9±0.08μs      14.3±0.03μs     0.96  core.expand.TimeExpand.time_expand_complex_number_1
       22.7±0.2μs       22.1±0.2μs     0.98  core.expand.TimeExpand.time_expand_complex_number_2
      22.9±0.09μs         23.5±1μs     1.03  core.expand.TimeExpand.time_expand_frac
       60.9±0.9μs       60.2±0.1μs     0.99  core.expand.TimeExpand.time_expand_log
      8.46±0.06μs      8.14±0.05μs     0.96  core.expand.TimeExpand.time_expand_modulus
       16.8±0.2μs       15.9±0.1μs     0.95  core.expand.TimeExpand.time_expand_negative_integers_powers
       16.3±0.3μs      15.7±0.06μs     0.96  core.expand.TimeExpand.time_expand_no_log
       22.0±0.4μs       21.5±0.1μs     0.98  core.expand.TimeExpand.time_expand_no_multinomial
       23.9±0.5μs       23.3±0.2μs     0.97  core.expand.TimeExpand.time_expand_non_commutative
         683±10ns          650±1ns     0.95  core.expand.TimeExpand.time_expand_nothing_todo
       23.0±0.1μs       23.2±0.1μs     1.01  core.expand.TimeExpand.time_expand_power_base
       32.1±0.2μs       31.0±0.5μs     0.97  core.expand.TimeExpand.time_expnad_radicals
      2.37±0.02ms      2.37±0.01ms     1.00  cse.TimeCSE.time_cse((2, 10))
      2.57±0.02ms      2.59±0.01ms     1.01  cse.TimeCSE.time_cse((2, 11))
      1.93±0.01ms       2.03±0.1ms     1.05  cse.TimeCSE.time_cse((2, 8))
      2.16±0.01ms      2.14±0.02ms     0.99  cse.TimeCSE.time_cse((2, 9))
      2.01±0.02ms      2.03±0.04ms     1.01  cse.TimeCSE.time_cse((3, 8))
      2.03±0.01ms      2.03±0.02ms     1.00  cse.TimeCSE.time_cse((4, 8))
          1.51±0s       1.54±0.01s     1.02  dsolve.TimeDsolve01.time_dsolve
        179±0.6ms        182±0.6ms     1.02  integrate.TimeIntegration01.time_doit
       76.3±0.3ms         78.5±1ms     1.03  integrate.TimeIntegration01.time_doit_meijerg
       36.1±0.6ms       36.0±0.2ms     1.00  integrate.TimeIntegrationRisch01.time_doit(1)
       52.7±0.5ms       52.5±0.3ms     1.00  integrate.TimeIntegrationRisch01.time_doit(10)
         644±20ms          652±4ms     1.01  integrate.TimeIntegrationRisch01.time_doit(100)
         39.7±2ms       36.0±0.7ms    ~0.91  integrate.TimeIntegrationRisch01.time_doit_risch(1)
       53.4±0.7ms       53.6±0.7ms     1.01  integrate.TimeIntegrationRisch01.time_doit_risch(10)
          644±7ms          650±8ms     1.01  integrate.TimeIntegrationRisch01.time_doit_risch(100)
      5.80±0.03μs      5.89±0.03μs     1.02  integrate.TimeIntegrationRisch02.time_doit(1)
        149±0.3ms        151±0.5ms     1.01  integrate.TimeIntegrationRisch02.time_doit(10)
        15.0±0.2s       15.0±0.06s     1.00  integrate.TimeIntegrationRisch02.time_doit(100)
       8.48±0.2μs      8.54±0.07μs     1.01  integrate.TimeIntegrationRisch02.time_doit_risch(1)
         149±20ms        149±0.4ms     1.00  integrate.TimeIntegrationRisch02.time_doit_risch(10)
       14.6±0.02s       15.0±0.04s     1.03  integrate.TimeIntegrationRisch02.time_doit_risch(100)
       63.0±0.1μs       62.6±0.2μs     0.99  integrate.TimeIntegrationRisch03.time_doit(1)
          218±2ms          223±2ms     1.02  integrate.TimeIntegrationRisch03.time_doit(10)
       3.05±0.06s       3.12±0.01s     1.02  integrate.TimeIntegrationRisch03.time_doit(100)
          795±3μs          803±5μs     1.01  integrate.TimeIntegrationRisch03.time_doit_risch(1)
          219±3ms        222±0.5ms     1.01  integrate.TimeIntegrationRisch03.time_doit_risch(10)
       3.04±0.02s       3.12±0.01s     1.03  integrate.TimeIntegrationRisch03.time_doit_risch(100)
       43.9±0.5ms       43.9±0.2ms     1.00  lambdify.TimeLambdifyCreation.time_lambdify_create(['math'])
      44.1±0.08ms       44.2±0.2ms     1.00  lambdify.TimeLambdifyCreation.time_lambdify_create(['numpy'])
      9.68±0.03μs      9.82±0.05μs     1.01  lambdify.TimeLambdifyEvaluate.time_lambdify_evaluate(['math'])
       96.8±0.6μs        102±0.2μs     1.05  lambdify.TimeLambdifyEvaluate.time_lambdify_evaluate(['numpy'])
            53.7M            53.5M     1.00  large_exprs.TimeLargeExpressionOperations.peakmem_jacobian_wrt_functions
            53.1M            52.8M     0.99  large_exprs.TimeLargeExpressionOperations.peakmem_jacobian_wrt_symbols
            53.1M            52.8M     0.99  large_exprs.TimeLargeExpressionOperations.peakmem_subs
       47.4±0.1ms      48.0±0.06ms     1.01  large_exprs.TimeLargeExpressionOperations.time_count_ops
      32.9±0.08ms      33.5±0.06ms     1.02  large_exprs.TimeLargeExpressionOperations.time_cse
      10.3±0.02ms       10.0±0.1ms     0.98  large_exprs.TimeLargeExpressionOperations.time_free_symbols
       1.60±0.01s       1.62±0.01s     1.01  large_exprs.TimeLargeExpressionOperations.time_jacobian_wrt_functions
       43.8±0.6ms      43.2±0.03ms     0.99  large_exprs.TimeLargeExpressionOperations.time_jacobian_wrt_symbols
       1.60±0.01s       1.62±0.02s     1.01  large_exprs.TimeLargeExpressionOperations.time_manual_jacobian_wrt_functions
          501±2ms          502±4ms     1.00  large_exprs.TimeLargeExpressionOperations.time_subs
       3.84±0.02s          3.86±0s     1.01  logic.LogicSuite.time_dpll
        190±0.9ms        181±0.8ms     0.95  logic.LogicSuite.time_dpll2
       36.7±0.3ms      36.5±0.08ms     0.99  logic.LogicSuite.time_load_file
        143±0.6μs          142±1μs     0.99  matrices.TimeDiagonalEigenvals.time_eigenvals
       9.89±0.1μs      9.94±0.05μs     1.00  matrices.TimeMatrixExpression.time_Add
        50.0±20μs       46.8±0.2μs     0.94  matrices.TimeMatrixExpression.time_MatAdd
       31.3±0.4ms      31.1±0.08ms     1.00  matrices.TimeMatrixExpression.time_MatAdd_doit
          537±3μs          522±3μs     0.97  matrices.TimeMatrixExpression.time_MatMul
      31.2±0.07ms      31.0±0.01ms     1.00  matrices.TimeMatrixExpression.time_MatMul_doit
      9.74±0.02μs      9.91±0.09μs     1.02  matrices.TimeMatrixExpression.time_Mul
          220±1μs          224±2μs     1.02  matrices.TimeMatrixGetItem.time_ImmutableDenseMatrix_getitem
          214±2μs          209±1μs     0.97  matrices.TimeMatrixGetItem.time_ImmutableSparseMatrix_getitem
          161±3μs        158±0.9μs     0.98  matrices.TimeMatrixGetItem.time_MutableDenseMatrix_getitem
          167±2μs          162±1μs     0.97  matrices.TimeMatrixGetItem.time_MutableSparseMatrix_getitem
       11.0±0.1ms      11.3±0.05ms     1.02  matrices.TimeMatrixPower.time_Case1
        134±0.9ms        136±0.5ms     1.02  matrices.TimeMatrixPower.time_Case2
          129±3ms         136±10ms     1.05  matrices.TimeMatrixPower.time_Case3
      9.66±0.08ms       9.81±0.1ms     1.02  physics.mechanics.kane.KanesMethodMassSpringDamper.time_kanesmethod_mass_spring_damper
       27.4±0.2ms       27.8±0.5ms     1.01  physics.mechanics.lagrange.LagrangesMethodMassSpringDamper.time_lagrangesmethod_mass_spring_damper
       7.50±0.02s        7.82±0.2s     1.04  polygon.PolygonArbitraryPoint.time_bench01
        378±0.6μs        395±0.9μs     1.05  polygon.PolygonAttributes.time_area
      1.85±0.01ms      1.89±0.01ms     1.03  polygon.PolygonAttributes.time_centroid
      6.24±0.05ms      6.22±0.02ms     1.00  polygon.PolygonAttributes.time_create
      1.58±0.01ms         1.54±0ms     0.97  polygon.PolygonAttributes.time_perimeter
         1.69±0ms      1.61±0.02ms     0.96  polygon.PolygonAttributes.time_sides
       1.25±0.01s          1.23±0s     0.99  polygon.PolygonCutSection.time_bench01
         116±10ms         147±20ms    ~1.27  polygon.PolygonDistance.time_bench01
        180±0.4ms         170±10ms     0.94  polygon.PolygonEnclosesPoint.time_bench01
          964±2ms         988±40ms     1.03  polygon.PolygonFirstMoment.time_bench01
         56.1±2ms       54.4±0.4ms     0.97  polygon.PolygonIsConvex.time_bench01
       7.35±0.6ms      7.01±0.02ms     0.95  polygon.PolygonPolarSecondMoment.time_bench01
       6.93±0.2ms      6.98±0.03ms     1.01  polygon.PolygonSecondMoment.time_bench01
      9.35±0.03ms      9.67±0.02ms     1.03  polygon.PolygonSectionModulus.time_bench01
        106±0.5μs        107±0.4μs     1.01  polys.TimePolyManyGens.time_create_poly(1)
        156±0.6μs        155±0.4μs     0.99  polys.TimePolyManyGens.time_create_poly(10)
      1.30±0.03ms      1.23±0.01ms     0.95  polys.TimePolyManyGens.time_create_poly(100)
       21.3±0.6ms       20.8±0.2ms     0.98  polys.TimePolyManyGens.time_create_poly(500)
       3.39±0.3μs      3.44±0.01μs     1.01  polys.TimePolyManyGens.time_is_linear(1)
       37.4±0.2μs       36.6±0.2μs     0.98  polys.TimePolyManyGens.time_is_linear(10)
        1000±70μs          937±4μs     0.94  polys.TimePolyManyGens.time_is_linear(100)
         19.9±2ms      19.5±0.07ms     0.98  polys.TimePolyManyGens.time_is_linear(500)
        146±0.2μs         154±20μs     1.05  solve.TimeMatrixArithmetic.time_dense_add(10, 0)
          160±2μs        163±0.5μs     1.02  solve.TimeMatrixArithmetic.time_dense_add(10, 5)
       20.2±0.1μs      20.5±0.05μs     1.02  solve.TimeMatrixArithmetic.time_dense_add(3, 0)
       26.2±0.3μs      25.9±0.08μs     0.99  solve.TimeMatrixArithmetic.time_dense_add(3, 5)
       30.7±0.1μs      31.6±0.05μs     1.03  solve.TimeMatrixArithmetic.time_dense_add(4, 0)
       38.5±0.2μs       38.4±0.2μs     1.00  solve.TimeMatrixArithmetic.time_dense_add(4, 5)
      57.8±0.07μs       60.2±0.1μs     1.04  solve.TimeMatrixArithmetic.time_dense_add(6, 0)
      67.6±0.09μs       69.4±0.1μs     1.03  solve.TimeMatrixArithmetic.time_dense_add(6, 5)
       2.21±0.1ms      2.24±0.01ms     1.02  solve.TimeMatrixArithmetic.time_dense_multiply(10, 0)
       2.70±0.1ms      2.76±0.01ms     1.02  solve.TimeMatrixArithmetic.time_dense_multiply(10, 5)
         83.0±1μs         87.5±3μs     1.05  solve.TimeMatrixArithmetic.time_dense_multiply(3, 0)
        145±0.3μs          148±1μs     1.02  solve.TimeMatrixArithmetic.time_dense_multiply(3, 5)
          177±1μs          188±1μs     1.06  solve.TimeMatrixArithmetic.time_dense_multiply(4, 0)
         354±60μs        329±0.8μs     0.93  solve.TimeMatrixArithmetic.time_dense_multiply(4, 5)
          524±6μs          549±3μs     1.05  solve.TimeMatrixArithmetic.time_dense_multiply(6, 0)
          841±2μs          867±7μs     1.03  solve.TimeMatrixArithmetic.time_dense_multiply(6, 5)
         185±10μs        181±0.4μs     0.98  solve.TimeMatrixOperations.time_det(3, 0)
         302±20μs         294±20μs     0.97  solve.TimeMatrixOperations.time_det(3, 2)
          283±3μs          281±9μs     0.99  solve.TimeMatrixOperations.time_det(3, 5)
          916±7μs         928±30μs     1.01  solve.TimeMatrixOperations.time_det(4, 0)
      6.03±0.08ms      6.08±0.03ms     1.01  solve.TimeMatrixOperations.time_det(4, 2)
      1.61±0.01ms      1.64±0.05ms     1.02  solve.TimeMatrixOperations.time_det(4, 5)
        181±0.6μs        180±0.4μs     1.00  solve.TimeMatrixOperations.time_det_bareiss(3, 0)
          291±1μs         300±10μs     1.03  solve.TimeMatrixOperations.time_det_bareiss(3, 2)
          281±8μs          281±2μs     1.00  solve.TimeMatrixOperations.time_det_bareiss(3, 5)
          903±5μs          922±5μs     1.02  solve.TimeMatrixOperations.time_det_bareiss(4, 0)
      6.05±0.02ms       6.30±0.2ms     1.04  solve.TimeMatrixOperations.time_det_bareiss(4, 2)
      1.60±0.01ms         1.62±0ms     1.01  solve.TimeMatrixOperations.time_det_bareiss(4, 5)
        179±0.5μs          182±5μs     1.02  solve.TimeMatrixOperations.time_det_berkowitz(3, 0)
        291±0.8μs        293±0.9μs     1.01  solve.TimeMatrixOperations.time_det_berkowitz(3, 2)
        279±0.7μs          279±2μs     1.00  solve.TimeMatrixOperations.time_det_berkowitz(3, 5)
      1.13±0.01ms      1.16±0.01ms     1.03  solve.TimeMatrixOperations.time_det_berkowitz(4, 0)
      1.39±0.01ms         1.40±0ms     1.01  solve.TimeMatrixOperations.time_det_berkowitz(4, 2)
      1.47±0.02ms         1.49±0ms     1.01  solve.TimeMatrixOperations.time_det_berkowitz(4, 5)
        431±0.9μs          434±1μs     1.01  solve.TimeMatrixOperations.time_rank(3, 0)
       77.8±0.3ms       78.4±0.5ms     1.01  solve.TimeMatrixOperations.time_rank(3, 2)
       37.7±0.2ms       38.5±0.5ms     1.02  solve.TimeMatrixOperations.time_rank(3, 5)
          697±3μs          715±3μs     1.03  solve.TimeMatrixOperations.time_rank(4, 0)
       85.0±0.3ms         84.6±2ms     1.00  solve.TimeMatrixOperations.time_rank(4, 2)
        87.1±10ms       84.9±0.1ms     0.97  solve.TimeMatrixOperations.time_rank(4, 5)
          222±6μs        229±0.5μs     1.03  solve.TimeMatrixOperations.time_rref(3, 0)
         45.8±8ms      41.3±0.06ms    ~0.90  solve.TimeMatrixOperations.time_rref(3, 2)
        90.7±10ms       79.3±0.2ms    ~0.87  solve.TimeMatrixOperations.time_rref(3, 5)
          463±3μs        489±0.9μs     1.06  solve.TimeMatrixOperations.time_rref(4, 0)
          103±5ms        104±0.8ms     1.01  solve.TimeMatrixOperations.time_rref(4, 2)
          116±1ms        117±0.3ms     1.01  solve.TimeMatrixOperations.time_rref(4, 5)
       59.3±0.6ms       60.2±0.3ms     1.01  solve.TimeMatrixPositiveDefinite.time_A_is_positive_definite
        63.2±20ms       59.8±0.1ms     0.95  solve.TimeMatrixPositiveDefinite.time_A_sym_is_positive_definite
         329±10ms          327±3ms     0.99  solve.TimeMatrixSolve.time_solve('ADJ')
         864±40ms          847±8ms     0.98  solve.TimeMatrixSolve.time_solve('GE')
      1.22±0.01ms      1.23±0.01ms     1.01  solve.TimeMatrixSolve.time_solve('LU')
         267±20ms          253±1ms     0.95  solve.TimeMatrixSolve.time_solve_sym('ADJ')
         743±40ms          728±9ms     0.98  solve.TimeMatrixSolve.time_solve_sym('GE')
       1.33±0.3ms         1.22±0ms     0.92  solve.TimeMatrixSolve.time_solve_sym('LU')
      2.38±0.01ms      2.39±0.01ms     1.00  solve.TimeMatrixSolve2.time_LDLsolve
      2.16±0.02ms      2.16±0.01ms     1.00  solve.TimeMatrixSolve2.time_cholesky_solve
      1.23±0.06ms         1.21±0ms     0.99  solve.TimeMatrixSolve2.time_lusolve
         711±20μs          704±2μs     0.99  solve.TimeMatrixSolvePyDyFast.time_LUsolve(1)
      1.24±0.05ms      1.22±0.01ms     0.98  solve.TimeMatrixSolvePyDyFast.time_LUsolve(2)
       2.15±0.1ms      2.09±0.01ms     0.97  solve.TimeMatrixSolvePyDyFast.time_LUsolve(3)
       1.33±0.2ms         1.26±0ms     0.95  solve.TimeMatrixSolvePyDySlow.time_LDLsolve(1)
       1.16±0.2ms         1.13±0ms     0.98  solve.TimeMatrixSolvePyDySlow.time_cholesky_solve(1)
         74.3±2ms       75.0±0.2ms     1.01  solve.TimeMatrixSolvePyDySlow.time_linsolve(1)
         77.0±2ms       76.4±0.2ms     0.99  solve.TimeMatrixSolvePyDySlow.time_solve(1)
          422±8μs          419±3μs     0.99  solve.TimeRationalSystem.time_linsolve(1)
       16.3±0.2ms      16.6±0.09ms     1.02  solve.TimeRationalSystem.time_linsolve(10)
      1.50±0.01ms      1.52±0.01ms     1.01  solve.TimeRationalSystem.time_linsolve(3)
      3.64±0.04ms      3.64±0.02ms     1.00  solve.TimeRationalSystem.time_linsolve(5)
       1.45±0.1ms      1.47±0.01ms     1.01  solve.TimeRationalSystem.time_solve(1)
         67.8±4ms       62.8±0.2ms     0.93  solve.TimeRationalSystem.time_solve(10)
       7.25±0.2ms      7.07±0.03ms     0.97  solve.TimeRationalSystem.time_solve(3)
       17.6±0.6ms      17.0±0.04ms     0.97  solve.TimeRationalSystem.time_solve(5)
      1.17±0.06ms          992±8μs    ~0.85  solve.TimeRationalSystemSymbol.time_linsolve(1)
      4.04±0.02ms      4.10±0.01ms     1.02  solve.TimeRationalSystemSymbol.time_linsolve(3)
       11.3±0.4ms       10.8±0.1ms     0.96  solve.TimeRationalSystemSymbol.time_linsolve(5)
       2.00±0.2ms      1.61±0.01ms    ~0.80  solve.TimeRationalSystemSymbol.time_solve(1)
         9.07±1ms      7.71±0.03ms    ~0.85  solve.TimeRationalSystemSymbol.time_solve(3)
       18.1±0.5ms      17.9±0.07ms     0.98  solve.TimeRationalSystemSymbol.time_solve(5)
       47.6±0.4ms      47.5±0.09ms     1.00  solve.TimeSolve01.time_solve
       48.2±0.6ms       47.5±0.1ms     0.99  solve.TimeSolve01.time_solve_nocheck
              n/a              n/a      n/a  solve.TimeSolveLinSys.time_solve_lin_sys
              n/a              n/a      n/a  solve.TimeSolveLinSys.time_to_expr_eqs
              n/a              n/a      n/a  solve.TimeSolveLinSys.time_verify_sol
              n/a              n/a      n/a  solve.TimeSolveLinSys10x8.time_solve_lin_sys
              n/a              n/a      n/a  solve.TimeSolveLinSys10x8.time_to_expr_eqs
              n/a              n/a      n/a  solve.TimeSolveLinSys10x8.time_verify_sol
              n/a              n/a      n/a  solve.TimeSolveLinSys165x165.time_solve_lin_sys
              n/a              n/a      n/a  solve.TimeSolveLinSys165x165.time_to_expr_eqs
              n/a              n/a      n/a  solve.TimeSolveLinSys165x165.time_verify_sol
         207±10ms        213±0.1ms     1.03  solve.TimeSolveLinSys189x49.time_solve_lin_sys
      80.2±0.09ms       81.5±0.6ms     1.02  solve.TimeSolveLinSys189x49.time_to_expr_eqs
          300±2ms        303±0.7ms     1.01  solve.TimeSolveLinSys189x49.time_verify_sol
      17.9±0.05ms      17.6±0.05ms     0.98  solve.TimeSolveSparsePolySystem.time_solve(1)
       39.1±0.1ms      35.7±0.08ms     0.91  solve.TimeSolveSparsePolySystem.time_solve(2)
-      69.4±0.4ms       53.9±0.2ms     0.78  solve.TimeSolveSparsePolySystem.time_solve(3)
      2.03±0.02ms         2.06±0ms     1.01  solve.TimeSparseSystem.time_linear_eq_to_matrix(10)
      6.20±0.03ms      6.32±0.01ms     1.02  solve.TimeSparseSystem.time_linear_eq_to_matrix(20)
      12.7±0.03ms      12.9±0.04ms     1.02  solve.TimeSparseSystem.time_linear_eq_to_matrix(30)
       30.0±0.2ms       30.8±0.4ms     1.03  solve.TimeSparseSystem.time_linsolve_Aaug(10)
       86.6±0.4ms       88.1±0.4ms     1.02  solve.TimeSparseSystem.time_linsolve_Aaug(20)
          172±2ms        174±0.2ms     1.01  solve.TimeSparseSystem.time_linsolve_Aaug(30)
       29.8±0.2ms       30.5±0.3ms     1.02  solve.TimeSparseSystem.time_linsolve_Ab(10)
       86.3±0.9ms       87.4±0.4ms     1.01  solve.TimeSparseSystem.time_linsolve_Ab(20)
        170±0.7ms        173±0.5ms     1.02  solve.TimeSparseSystem.time_linsolve_Ab(30)
      13.7±0.04ms      14.2±0.01ms     1.04  solve.TimeSparseSystem.time_linsolve_eqs(10)
       43.6±0.2ms       45.1±0.1ms     1.03  solve.TimeSparseSystem.time_linsolve_eqs(20)
       89.8±0.5ms       93.0±0.1ms     1.04  solve.TimeSparseSystem.time_linsolve_eqs(30)
      34.3±0.04ms       34.0±0.2ms     0.99  solve.TimeSparseSystem.time_solve(10)
-         184±8ms       67.9±0.3ms     0.37  solve.TimeSparseSystem.time_solve(20)
-         362±1ms        102±0.1ms     0.28  solve.TimeSparseSystem.time_solve(30)
        211±0.5ms        209±0.9ms     0.99  sum.TimeSum.time_doit

```